#LT2022-91 Integrate MYSQL DB to Java source code

EXR-38 opened the pull request:
Removed unused StringBuilder object.
Addressed the infinite loop scenario by altering loop structure.
Addressed the scenario when user inputs more than two arguments.
Addressed the situation when user inputs some special character with the search word.

EXR-38 committed:
WordSearchV9.0

Removed unused StringBuilder object.
Addressed the infinite loop scenario by altering loop structure.
Addressed the scenario when user inputs more than two arguments.
Addressed the situation when user inputs some special character with the search word.

kssreejith1994 commented on Main.java:
```diff
@@ -4,26 +4,42 @@
-    final static String SINGLE_SPACE = "";
+    static final String FILE_TYPE_TXT = ".txt";
+    static final String FILE_TYPE_JSON = ".json";
+    static final String REMOVE_SPECIAL_CHAR = "\\p{Punct}";
```
How is REMOVE_SPECIAL_CHAR and EXCLUDE_SPECIAL_CHAR different?

EXR-38 replied:
They are the same actually. EXCLUDE_SPECIAL_CHAR is not used anywhere in the program. I'll remove it.

kssreejith1994 commented on Main.java:
```diff
@@ -4,26 +4,42 @@
     public static void main(String[] args) {
 
-        if (args.length != 2) {
+        if (args.length < 2) {
```
I believe you don't need to check less than and greater than 2 args separately, rather check for !equal to 2 and print message like Program accepts only two arguments-filepath and search word.. something like this.

EXR-38 replied:
Sure. I will do that.

kssreejith1994 commented on Main.java:
```diff
@@ -4,26 +4,42 @@
+        }
         String textFilePath = args[0];
         String searchKeyword = args[1];
+        Matcher matcher = Pattern.compile(REMOVE_SPECIAL_CHAR).matcher(searchKeyword);
```
Why is punctuation check made on user input? We can just return the searchword not found right? User can pass anything as input.

EXR-38 replied:
Even if the user inputs something like "@vibecheck" and there exists a word "@vibecheck" in the file, our code will only return "Word does not exist" since we are removing all the special characters from the file contents. Hence the punctuation check.

kssreejith1994 replied:
Got it, that makes sense, but the same applies if the file contains a word like say 'hi-tech' in the file and user search for hi-tech, but will not find it since hi-tech will be read as hi tech in the program right? Or hyphen words are not separated?

EXR-38 replied:
Got it. I will alter the code to not separate hyphen words.

kssreejith1994 commented:
Please edit the PR name to appropriate one and should have proper commit message. Also, please follow branching guidelines from next story.

EXR-38 committed:
Add support for hyphen words. Integrate MySQL DB.

Added support for hyphened words during word search.
Integrated MySQL DB to the java source code.
